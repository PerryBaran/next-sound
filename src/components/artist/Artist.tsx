import Image from "next/image"
import css from "./artist.module.css"
import { profile } from "../../media/icons"
import Link from "next/link"

interface Props {
  name: string
  image?: string
}

export default function Artist({ name, image }: Props) {
  return (
    <div className={css["container"]}>
      <Link href={`/profile/${name}`} className={css["link"]}>
        <Image
          src={image || profile}
          alt={`album by ${name}`}
          height={100}
          width={100}
        />
        <h2 className={css["name"]}>{name}</h2>
      </Link>
    </div>
  )
}
