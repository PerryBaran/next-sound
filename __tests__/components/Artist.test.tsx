import { render, screen } from "@testing-library/react"
import Artist from "../../src/components/artist/Artist"

describe("Artist", () => {
  const props = {
    name: "name",
    image: "/image"
  }

  test("renders correctly", () => {
    render(<Artist {...props} />)

    expect(screen.getByRole("link")).toHaveAttribute(
      "href",
      `/profile/${props.name}`
    )
    expect(screen.getByAltText(`album by ${props.name}`)).toHaveAttribute(
      "src",
      `/_next/image?url=%2F${props.image.substring(
        1,
        props.image.length
      )}&w=256&q=75`
    )
    expect(screen.getByRole("heading")).toHaveTextContent(props.name)
  })

  test("snapshot", () => {
    const { asFragment } = render(<Artist {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
