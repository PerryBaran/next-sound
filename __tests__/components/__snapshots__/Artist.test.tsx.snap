// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <a
      class="link"
      href="/profile/name"
    >
      <img
        alt="album by name"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=%2Fimage&w=256&q=75"
        srcset="/_next/image?url=%2Fimage&w=128&q=75 1x, /_next/image?url=%2Fimage&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
      <h2
        class="name"
      >
        name
      </h2>
    </a>
  </div>
</DocumentFragment>
`;
